ros/lib/rbac_interface.py

```diff
@@ -118,3 +120,79 @@ def _is_mgmt_url(path):
     Small helper to test if URL is for management API.
     """
     return path.startswith("/mgmt/")
+
+
+def set_host_groups(rbac_response):
+    """
+    We now also have to store the host group information we get from inventory.
+    This comes in the resource definition within the RBAC response:
+
+    {
+      "resourceDefinitions": [
+        {
+          "attributeFilter": {
+            "key": "inventory.groups",
+            "value": [
+              "group 1",
+              "group 2"
+            ],
+            "operation": "in"
+          }
+        }
+      ],
+      "permission": "inventory:hosts:read"
+    }
+
+    If the permission is 'inventory:hosts:read', we find one of the resource
+    definitions that has an attribute filter key of 'inventory.groups', and
+    we get the list of inventory groups from its value. We currently ignore
+    the operation value.
+    """
+
+    if rbac_response is None:
+        return  # we can't store any host groups on None
+
+    if 'data' not in rbac_response:
+        LOG.info("Warning: The response from RBAC does not contain 'data' list to fetch group details")
+        return
+
+    role_list = rbac_response['data']
+
+    for role in role_list:
+        if 'permission' not in role:
+            continue
+        if role['permission'] != 'inventory:hosts:read':
+            continue
+        # ignore the failure modes, try moving on to other roles that
+        # also match this permission
+        if 'resourceDefinitions' not in role:
+            continue
+        if not isinstance(role['resourceDefinitions'], list):
+            continue
+        for rscdef in role['resourceDefinitions']:
+            if not isinstance(rscdef, dict):
+                continue
+            if 'attributeFilter' not in rscdef:
+                continue
+            attrfilter = rscdef['attributeFilter']
+            if not isinstance(attrfilter, dict):
+                continue
+            if 'key' not in attrfilter and 'value' not in attrfilter:
+                continue
+            if attrfilter['key'] != 'group.id':
+                continue
+            value = attrfilter['value']
+            # Early versions of the spec say the value is a list; later
+            # versions say it's a string with a JSON-encoded list.  Let's try
+            # to cope with the latter by converting it into the former.
+            if isinstance(value, str) and value[0] == '[' and value[-1] == ']':
+                value = json.loads(value)
+            if not isinstance(value, list):
+                continue
+            # Finally, we have the right key: set its value (a list) as
+            # the 'host_group_attr' property
+            # Maybe it would be good to set values as json here instead of list
+            setattr(request, host_group_attr, value)
```

Comment:
@upadhyeammit I am just curious to know as I have limited knowledge of python that are we setting a att for any request we receive on the backend side for groups? and then we will access the group info in the query?

Reply:
Yea you are correct. We are setting an extra attribute on request object! initially my thoughts were somehow returning or exposing this group list using var or method for each request, however patching a new attribute on existing object felt better in terms of understanding, 
1. Its per request group list
2. Carry it cleanly with each request object

Reply:
@upadhyeammit got it thanks for sharing the details 🙌🏻